File: algorithms/ar-ssrt/python3/selection_sort.py
def selection(arr):
    for i in range(len(arr)):
        min_idx = i
        for j in range(i + 1, len(arr)):
            if arr[min_idx] > arr[j]:
                min_idx = j

        arr[i], arr[min_idx] = arr[min_idx], arr[i]

    print("Sorted array")
    for i in range(len(arr)):
        print("%d" % arr[i]),


a = [1, 88, 2, 56, 56, 5, 56, 2, 5, 65, 8]

print(selection(a))

File: algorithms/ar-ssrt/java/SelectionSort.java
class SelectionSort {
    // Driver code to test above
    public static void main(String args[]) {
        SelectionSort ob = new SelectionSort();
        Scanner s = new Scanner(System.in);
        System.out.println("Insert array elements seperating with spaces only");

        String temp = s.nextLine();

        String ar[] = temp.split(" ");
        int arr[] = new int[ar.length];

        for (int i = 0; i < ar.length; i++) {
            arr[i] = Integer.parseInt(ar[i]);
        }

        ob.sort(arr);
        System.out.println("Sorted array");
        ob.printArray(arr);
    }

    public void sort(int arr[]) {
        int n = arr.length;

        // One by one move boundary of unsorted subarray
        for (int i = 0; i < n - 1; i++) {
            // Find the minimum element in unsorted array
            //int min_idx = i;
            int min = arr[i];
            int index = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[j] < min) {
                    min = arr[j];
                    index = j;
                }
            }

            // Swap the found minimum element with the first
            // element
            int temp = arr[index];
            arr[index] = arr[i];
            arr[i] = temp;
        }
    }

    // Prints the array
    public void printArray(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n; ++i)
            System.out.print(arr[i] + " ");
        System.out.println();
    }
}
